.github/actions/commit_and_push/commit_and_push.ts

```diff
@@ -0,0 +1,187 @@
+import { execSync } from 'child_process';
+import { readFile, writeFile } from 'fs/promises';
+import { tmpdir } from 'os';
+import { join } from 'path';
+
+type CreateCommitInput = {
+    branch: {
+        repositoryNameWithOwner: string;
+        branchName: string;
+    };
+    fileChanges: {
+        additions: Addition[];
+    };
+    message: {
+        headline: string;
+    };
+    expectedHeadOid: string;
+};
+
+type Addition = {
+    path: string;
+    contents: string;
+};
+
+async function commitAndPush(): Promise<void> {
+    verifyEnvironmentVariables();
+
+    execSync(`git add ${process.env.ADD_FILES || '--all'}`, {
+        stdio: 'inherit'
+    });
+
+    const changedFiles = getChangedFiles();
+
+    if (changedFiles.length === 0) {
+        console.info('ℹ️ No changes to commit');
+        return;
+    }
+    console.info(`📝 Found ${changedFiles.length} changed files:`);
+    changedFiles.forEach((file) => console.info(`  - ${file}`));
+
+    const expectedHeadOid = getExpectedHeadOid();
+
+    const additions = await encodeFileContents(changedFiles);
+
+    const input = createInput(additions, expectedHeadOid);
+
+    await createCommitWithRetry(input);
+
+    console.info('✅ Commit created successfully!');
+}
+
+function verifyEnvironmentVariables(): void {
+    // Validate required environment variables
+    const requiredEnvVars = [
+        'GH_TOKEN',
+        'BRANCH_NAME',
+        'COMMIT_MESSAGE',
+        'GITHUB_REPOSITORY'
+    ];
+    for (const envVar of requiredEnvVars) {
+        if (!process.env[envVar]) {
+            throw new Error(`Missing required environment variable: ${envVar}`);
+        }
+    }
+}
+
+function getChangedFiles(): string[] {
+    const changedFilesOutput = execSync('git diff --cached --name-only', {
+        encoding: 'utf8'
+    });
+    const changedFiles = changedFilesOutput
+        .trim()
+        .split('\n')
+        .filter((f) => f.length > 0);
+
+    return changedFiles;
+}
+
+function getExpectedHeadOid(): string {
+    return execSync('git rev-parse HEAD', {
+        encoding: 'utf8'
+    }).trim();
+}
+
+async function encodeFileContents(changedFiles: string[]): Promise<Addition[]> {
+    return Promise.all(
+        changedFiles.map(async (file) => {
+            try {
+                const content = (await readFile(file)).toString('base64');
+                return {
+                    path: file,
+                    contents: content
+                };
+            } catch (error) {
+                throw new Error(`Failed to read file ${file}: ${error}`);
+            }
+        })
+    );
+}
+
+function createInput(
+    additions: Addition[],
+    expectedHeadOid: string
+): CreateCommitInput {
+    return {
+        branch: {
+            repositoryNameWithOwner: process.env.GITHUB_REPOSITORY!,
```

Comment:
No `!`

Reply:
Instead create the variables in `verifyEnvironmentVariables` and pass them down into the functions that need them.

Reply:
done